k8s-tester/secureCN/tester.go

```diff
@@ -0,0 +1,257 @@
+	if err != nil {
+		return fmt.Errorf("failed to execute command. command output: %s. %v", out, err)
+	}
+
+	waitForAgentArgs := []string{
+		ts.cfg.Client.Config().KubectlPath,
+		"--kubeconfig=" + ts.cfg.Client.Config().KubeconfigPath,
+		"wait",
+		"-n",
+		"portshift",
+		"pod",
+		"-l",
+		"app=portshift-agent",
+		"--for",
+		"condition=ready",
+		"--timeout",
+		"10m",
+	}
+	out, err = exec.New().CommandContext(context.TODO(), waitForAgentArgs[0], waitForAgentArgs[1:]...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to execute command. command output: %s. %v", out, err)
+	}
+
+	ts.cfg.Logger.Info("Successfully installed securecn agent")
+	return nil
+}
+
+func cleanInstallationFiles() {
+	_ = os.Remove(SecureCNAGBundleName)
+	_ = os.Remove(SecureCNIstioAGBundleName)
+	_ = os.Remove(SecureCNInstallBundle)
+	_ = os.Remove(SecureCNBundleYaml)
+}
+
+func downloadAndExtractBundle(clientWrapper api_client.HttpClientWrapper, clusterID strfmt.UUID) error {
+	f, err := os.Create(SecureCNBundleTarGz)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %v. %v", SecureCNBundleTarGz, err)
+	}
+	buffer := new(bytes.Buffer)
+	// download bundle
+	err = clientWrapper.EscherClient.DownloadKubernetesSecureCNBundle(context.TODO(), clientWrapper.HttpClient, buffer, clusterID)
+	_, err = io.Copy(f, buffer)
+	if err != nil {
+		return fmt.Errorf("failed to download bundle. %v", err)
+	}
+
+	open, err := os.Open(SecureCNBundleTarGz)
+	if err != nil {
+		return fmt.Errorf("failed to open file: %v. %v", SecureCNBundleTarGz, err)
+	}
+	defer os.Remove(SecureCNBundleTarGz)
+
+	err = utils.ExtractTarGz(open)
+	if err != nil {
+		return fmt.Errorf("failed to extract tar gz. %v", err)
+	}
+	return nil
+}
+
+func (ts *tester) Delete() error {
+	defer os.Remove(SecureCNUninstallScript)
+
+	ts.cfg.Logger.Info("Uninstalling securecn")
+
+	getUninstallerArgs := []string{
+		ts.cfg.Client.Config().KubectlPath,
+		"--kubeconfig=" + ts.cfg.Client.Config().KubeconfigPath,
+		"get",
+		"cm",
+		"-n",
+		"portshift",
+		"portshift-uninstaller",
+		"-o",
+		"jsonpath='{.data.config}'",
+	}
+	out, err := exec.New().CommandContext(context.TODO(), getUninstallerArgs[0], getUninstallerArgs[1:]...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to execute command. command output: %s. %v", out, err)
+	}
+
+	f, err := os.Create(SecureCNUninstallScript)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %v. %v", SecureCNUninstallScript, err)
+	}
+
+	outStr := string(out)
+
+	outStr = strings.TrimPrefix(outStr, "'")
+	outStr = strings.TrimSuffix(outStr, "'")
+
+	_, err = f.Write([]byte(outStr))
+
+	if err := f.Chmod(0700); err != nil {
+		return err
+	}
+	out, err = exec.New().CommandContext(context.TODO(), "./"+SecureCNUninstallScript).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to execute command. command output: %s. %v", out, err)
```

Comment:
can you add a context to each such print (e.g. failed to SecureCN uninstall command)